lib/pbench/test/unit/agent/test_validate_hostname.py

```diff
@@ -0,0 +1,15 @@
+"""Tests for validate_hostname
+"""
+
+from pbench.agent.utils import validate_hostname
+
+
+def test_validate_hostname():
+    assert validate_hostname("test") == 0
+    assert validate_hostname("test.example.com") == 0
+    assert validate_hostname("tes_t.example.com") == 1
+    assert validate_hostname("test-.example.com") == 1
+    assert validate_hostname("--run-label__") == 1
+    assert validate_hostname("127.0.0.1") == 0
+    assert validate_hostname("1270.0.0.1") == 0
+    assert validate_hostname("2001:0db8:85a3:0000:0000:8a2e:0370:7334") == 0
```

Comment:
I think there might be a small crack here that we need to fill in, either in this PR or in #2181 (or, if necessary, in a follow-on....).

I believe that the conventional way to specify an IPv6 address with a port is to put the node portion in square brackets:  e.g., `[::1]:8080`.

So, when someone tries `pbench-tool-meister-start --redis-server [::1]:17001`, the code is going to grab the brackets as part of the hostname.  However, `ipaddress.ip_address()` doesn't accept them.

So, we'll either need `validate_hostname()` to accept (and remove) them, or we'll need need its various callers (`BaseServer.__init__()` and others) to recognize and strip them.

Reply:
We don't any support for port numbers being specified in the current interfaces as of this PR, so it'll have to be addressed in PR #2181.